eq-author/src/components/PositionModal/index.js

```diff
@@ -32,64 +32,123 @@ const Trigger = styled.button.attrs({ type: "button" })`
   }
 `;
 
+const Indent = styled(Option)`
+  margin-left: ${({ indent }) => (indent ? 1 : 0)}em;
+`;
+
 const PositionModal = ({ options, onMove, selected }) => {
+  const positionButtonId = uniqueId("PositionModal");
+  const [isOpen, setIsOpen] = useState(false);
   const previousPosition = useMemo(
     () => options.findIndex(({ id }) => id === selected.id),
     [options, selected]
   );
-  const [isOpen, setIsOpen] = useState(false);
-  const [selectedPosition, setSelectedPosition] = useState(previousPosition);
-  const [positionButtonId] = useState(uniqueId("PositionModal"));
+
+  const [{ position, item, isSame }, setOption] = useState({
+    position: previousPosition,
+    item: options[previousPosition],
+  });
 
   const orderedOptions = options.filter(({ id }) => id !== selected.id);
-  orderedOptions.splice(selectedPosition, 0, selected);
+  selected.parentEnabled = item?.parentEnabled;
+  orderedOptions.splice(position, 0, selected);
 
   const handleClose = () => {
     setIsOpen(false);
-    setSelectedPosition(previousPosition);
+    setOption({
+      position: previousPosition,
+      item: orderedOptions[previousPosition],
+    });
   };
 
-  const handleOpen = () => setIsOpen(true);
-
-  const handleChange = ({ value }) => setSelectedPosition(parseInt(value, 10));
+  const handleChange = ({ value }) =>
+    setOption(({ item: previousItem }) => {
+      let count =
+        orderedOptions[value].type === "Folder" && value - position >= 0
+          ? orderedOptions.filter(
+              ({ parentId }) => parentId === orderedOptions[value].id
+            ).length
+          : 0;
+      return {
+        position: parseInt(value, 10) + count,
+        item: orderedOptions[value],
+        isSame: previousItem?.id === orderedOptions[value]?.id,
+      };
+    });
 
   const handleConfirm = (e) => {
     e.preventDefault();
-    setIsOpen(false);
+    const { parentId, position: itemPosition, parentEnabled } = item;
+    const lastIndex = orderedOptions.findIndex(({ id }) => item.id === id);
+
+    const checkIndex = position - lastIndex > -1;
+
+    const differentSection =
+      selected?.section?.id !== orderedOptions[position]?.sectionId;
+
+    const intoFolder =
+      !orderedOptions[previousPosition]?.parentEnabled && parentEnabled;
+    const outOfFolder =
+      orderedOptions[previousPosition]?.parentEnabled && !parentEnabled;
+    const fromFolderToFolder =
+      orderedOptions[previousPosition]?.parentEnabled && parentEnabled;
+
+    const inOrOut = intoFolder || outOfFolder;
+
+    const sameMove = checkIndex ? itemPosition + 1 : itemPosition - 1;
+    const folderMove = checkIndex ? itemPosition + 1 : itemPosition;
+
+    const positionCalculation = isSame
+      ? sameMove
+      : differentSection
+      ? folderMove
+      : fromFolderToFolder
+      ? folderMove
+      : inOrOut
+      ? folderMove
+      : itemPosition;
```

Comment:
Could this be changed to not use short hand, please? It's difficult to understand and I think we would benefit from having it use the tranditional if statements instead for if/when we read back over this.